plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -99,47 +102,61 @@ def main():
             application_id=dict(required=False),
             changelog=dict(required=False),
             description=dict(required=False),
-            revision=dict(required=False),
+            revision=dict(required=True),
             user=dict(required=False),
             appname=dict(required=False),
```

Comment:
```suggestion
            appname=dict(required=False, removed_in_version='4.0.0', removed_from_collection='community.general'),
```
Same for the other deprecated option(s).